src/robot.cpp

```diff
@@ -200,6 +217,22 @@ void Robot::WaitUntilReady()
             std::this_thread::yield();
         }
     }
+
+    if (HasError()) {
+        if (robot_if->IsTimeout())
+        {
+            throw std::runtime_error("Timeout during Robot::WaitUntilReady().");
+        } else {
+            throw std::runtime_error("Error during Robot::WaitUntilReady().");
+        }
+    }
+}
+
+void Robot::Initialize()
+{
+    Start();
+    WaitUntilReady();
+    RunCalibration();
```

Comment:
So when the executable of the robot is launch, the calibration is run right away?
We can try.

Reply:
Is that compatible with the DGM way of running?

Reply:
I guess yes because Start, WaitUntilReady and RunCalibration are public anyway :thinking:

Reply:
Yes. The public API is not changed. This is just a new convenience method to get the python example look a bit shorter (Thomas said a PD controller should take only 10 lines, and with this change it does as you don't need to call Start, WaitUntilReady, RunCalibration manyally, which would yield 12 lines ^^). This means our way of running the code to trigger the calibration manually is working still just fine (I checked it).